### PR TITLE
fix(kdropdownitem): attribute handling

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Bind attributes to underlying `.dropdown-item-trigger` instead of wrapper `li` element in `KDropdownItem`

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

File: src/components/KDropdown/KDropdownItem.vue [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
     class="k-dropdown-item"
-    :class="{
+    :class="[{
       'has-divider': hasDivider,
       'disabled': disabled,
       'danger': danger || isDangerous,
-      'dropdown-selected-option': selected
-    }"
+      'dropdown-selected-option': selected,
+    }, $attrs.class]"
```
Comment (adamdehaven) [inline src/components/KDropdown/KDropdownItem.vue:9]: question: should we purposefully bind inline styles to the parent as well? I don't think they should fallthrough

Comment (portikM) [reply src/components/KDropdown/KDropdownItem.vue:9]: I'm fine with whichever we pick

Comment (kaiarrowood) [inline src/components/KDropdown/KDropdownItem.vue:9]: I'm confused, wasn't the point to bind the `data-testid` to the element that actually triggers the click (ie. the `button`, `li`, etc)? If the child `<component>` still has a `data-testid` (which it does - `dropdown-item-trigger`) won't we still have the original problem attempting to catch the clicks in DD?

Comment (portikM) [reply src/components/KDropdown/KDropdownItem.vue:9]: No - if we provide `data-testid` it will override the original value

Comment (kaiarrowood) [reply src/components/KDropdown/KDropdownItem.vue:9]: But how? I'm not understanding what this `availableComponent` thing is

Comment (portikM) [reply src/components/KDropdown/KDropdownItem.vue:9]: It binds provided attributes after it sanitizes them a bit: https://github.com/Kong/kongponents/blob/75576a48ae122e9be72d891638d00b925289f3cf/src/components/KDropdown/KDropdownItem.vue#L156

Comment (portikM) [reply src/components/KDropdown/KDropdownItem.vue:9]: Going on with your suggestion to bind inline styles to the parent

File: src/components/KDropdown/KDropdownItem.vue [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
     class="k-dropdown-item"
-    :class="{
+    :class="[{
       'has-divider': hasDivider,
       'disabled': disabled,
       'danger': danger || isDangerous,
-      'dropdown-selected-option': selected
-    }"
+      'dropdown-selected-option': selected,
+    }, $attrs.class, $attrs.style]"
```
Comment (adamdehaven) [inline src/components/KDropdown/KDropdownItem.vue:9]: question: why are we moving all of the attributes? This is a larger breaking change

File: src/components/KDropdown/KDropdownItem.vue [hunk 3]
```diff
@@ -24,11 +24,17 @@
   </li>
 </template>
 
-<script lang="ts" setup>
+<script lang="ts">
 import type { DropdownItem, DropdownItemRenderedRecord, DropdownItemRenderedType, DropdownItemType } from '@/types'
 import type { PropType } from 'vue'
 import { computed, useAttrs } from 'vue'
 
+export default {
```
Comment (adamdehaven) [inline src/components/KDropdown/KDropdownItem.vue:32]: suggestion: if you need this you can use defineOptions

Comment (portikM) [reply src/components/KDropdown/KDropdownItem.vue]: Done

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**